roles/kubernetes/secrets/tasks/gen_certs_script.yml

```diff
@@ -47,24 +54,27 @@
   delegate_to: "{{groups['kube-master'][0]}}"
   when: gen_certs|default(false)
 
-- name: Gen_certs | run cert generation script
-  command: "{{ kube_script_dir }}/make-ssl.sh -f {{ kube_config_dir }}/openssl.conf -d {{ kube_cert_dir }}"
+- name: Gen_certs | run master cert generation script
+  command: "{{ kube_script_dir }}/make-ssl.sh -f {{ kube_config_dir }}/openssl-main.conf -d {{ kube_cert_dir }}"
   environment:
     - MASTERS: "{% for m in groups['kube-master'] %}
```

Comment:
You would maybe want to do the same here, to only add the hosts only ip with: `"{{ inventory_hostname}}"` like you do with the task `Gen_certs | run nodes cert generation script`

Reply:
This would  be  a more disruptive change. make_ssl.sh does iterate through the nodes and creates certificates specific to each node so the changes done to the nodes work  accordingly. But with the masters there are  certificates that are shared/the same for every master ( kube-apiserver, kube-controller-manager ) and certificates that are per master ( admin-nodeX )